### DISCUSSION TITLE
Is it normal for go-redis to take up ~1mb per subscription?

### DISCUSSION THREAD

**Discussion by cgvalayev** [redis/go-redis#2535] [Q&A]
We create a new redis subscription to a unique channel every time there is a new user socket connection. What we noticed is that memory of our service started to grow really fast with each connection. Is there a configuration that we are missing?

**Comment by monkey92t** [comment]
We haven't tracked the memory usage for each network connection, but we haven't received any user feedback about memory leaks or excessive memory usage caused by network connections.

If each connection occupies 1MB of memory, it doesn't seem to be a problem because we only need a few dozen network connections when using go-redis.

**Comment by cgvalayev** [comment]
You are correct, what we ended up realizing is that we were creating a brand new redis pubsub connection for each socket connection instead of reusing it.
